ChatSecure/Classes/View Controllers/OTRRoomOccupantsViewController.swift

```diff
@@ -10,89 +10,221 @@ import Foundation
+    public func setupViewHandler(databaseConnection:YapDatabaseConnection, roomKey:String) {
+        self.roomKey = roomKey
+        databaseConnection.read({ (transaction) in
+            if let key = self.roomKey {
+                self.room = OTRXMPPRoom.fetchObject(withUniqueID: key, transaction: transaction)
+            }
+        })
+        viewHandler = OTRYapViewHandler(databaseConnection: databaseConnection)
+        if let viewHandler = self.viewHandler, let roomKey = self.roomKey {
+            viewHandler.delegate = self
+            viewHandler.setup(DatabaseExtensionName.groupOccupantsViewName.name(), groups: [roomKey])
+        }
     }
     
-    private func setupTableView() {
-        self.view.addSubview(self.tableView)
-        self.tableView.autoPinEdgesToSuperviewEdges()
+    override open func viewDidLoad() {
+        super.viewDidLoad()
+        
+        self.headerRows.append("name")
+        self.headerRows.append("share")
+        self.headerRows.append("addFriends")
+        self.headerRows.append("mute")
+        self.headerRows.append("members")
+        self.footerRows.append("leave")
+        
+        let image = OTRGroupAvatarGenerator.avatarImage(withUniqueIdentifier: self.roomKey!, width: Int(largeAvatarView.frame.width), height: Int(largeAvatarView.frame.height))
+        largeAvatarView.image = image
+
+        //Setup Table View
+        if (self.tableView == nil) {
+            self.tableView = UITableView(frame: CGRect.zero, style: .plain)
+            self.view.addSubview(self.tableView)
+            self.tableView.autoPinEdgesToSuperviewEdges()
+        }
         self.tableView.dataSource = self
         self.tableView.delegate = self
-        self.tableView.register(UITableViewCell.self, forCellReuseIdentifier: OTRRoomOccupantsViewController.CellIdentifier)
+        self.tableView.register(OTRBuddyInfoCell.self, forCellReuseIdentifier: OTRRoomOccupantsViewController.CellIdentifier)
     }
-
-    override open func viewDidLoad() {
-        super.viewDidLoad()
-        setupTableView()
+    
+    open func createHeaderCell(indexPath:IndexPath, type:String) -> UITableViewCell {
+        switch type {
+        case "name":
+            let cell = tableView.dequeueReusableCell(withIdentifier: "cellGroupName", for: indexPath)
+            if let key = self.roomKey {
+                OTRDatabaseManager.shared.readOnlyDatabaseConnection?.read({ (transaction) in
+                    if let room = OTRXMPPRoom.fetchObject(withUniqueID: key, transaction: transaction) {
+                        cell.textLabel?.text = room.subject
+                        cell.detailTextLabel?.text = "" // Do we have creation date?
+                    }
+                })
+            }
+            cell.selectionStyle = .none
+            return cell
+        case "share":
+            let cell = tableView.dequeueReusableCell(withIdentifier: "cellGroupShare", for: indexPath)
```

Comment:
In my first approach I had a mapping string->cellIdentifier but since we have to do some configuration of the cells anyway (currently only for the "group name" cell) I felt that the mapping was pretty unnecessary and that the cellForRow... could be responsible for fetching the right type of cell.